driver-mapping/src/main/java/com/datastax/driver/mapping/MappingManager.java

```diff
@@ -74,6 +77,74 @@ public MappingManager(Session session, ProtocolVersion protocolVersion) {
+
+            @Override
+            public void onTableRemoved(TableMetadata table) {
+                synchronized (mappers) {
+                    Iterator<Mapper<?>> it = mappers.values().iterator();
+                    while (it.hasNext()) {
+                        Mapper<?> mapper = it.next();
+                        if (mapper.getTableMetadata().equals(table)) {
+                            LOGGER.error("Table {} has been removed; existing mappers for @Entity annotated {} will not work anymore", table.getName(), mapper.getMappedClass());
+                            it.remove();
+                        }
+                    }
+                }
+            }
+
+            @Override
+            public void onTableChanged(TableMetadata current, TableMetadata previous) {
+                synchronized (mappers) {
+                    Iterator<Mapper<?>> it = mappers.values().iterator();
+                    while (it.hasNext()) {
+                        Mapper<?> mapper = it.next();
+                        if (previous.equals(mapper.getTableMetadata())) {
+                            LOGGER.warn("Table {} has been altered; existing mappers for @Entity annotated {} might not work properly anymore",
+                                    previous.getName(), mapper.getMappedClass());
+                            it.remove();
+                        }
+                    }
+                }
+            }
+
+            @Override
+            public void onUserTypeRemoved(UserType type) {
+                synchronized (udtCodecs) {
+                    Iterator<MappedUDTCodec<?>> it = udtCodecs.values().iterator();
+                    while (it.hasNext()) {
+                        MappedUDTCodec<?> codec = it.next();
+                        if (codec.accepts(type)) {
+                            LOGGER.error("User type {} has been removed; existing mappers for @UDT annotated {} will not work anymore",
+                                    type, codec.getUdtClass());
+                            it.remove();
+                        }
+                    }
+                }
+            }
+
+            @Override
+            public void onUserTypeChanged(UserType current, UserType previous) {
+                synchronized (udtCodecs) {
+                    Iterator<MappedUDTCodec<?>> it = udtCodecs.values().iterator();
+                    while (it.hasNext()) {
+                        MappedUDTCodec<?> codec = it.next();
+                        if (previous.equals(codec.getCqlType())) {
+                            LOGGER.warn("User type {} has been altered; existing mappers for @UDT annotated {} might not work properly anymore",
+                                    previous, codec.getUdtClass());
+                            it.remove();
+                            // try to register an updated version of the previous codec
+                            try {
+                                getUDTCodec(codec.getUdtClass());
```

Comment:
This mutates the map. Aren't we risking a `ConcurrentModificationException` here? We could memorize the codecs to update and do it outside of the iteration.